tests/test_converters.py

```diff
@@ -18,90 +18,127 @@
 class TestNetworkxConverter(unittest.TestCase):
     def test_undirected_gnm_graph(self):
         g = networkx.gnm_random_graph(10, 10, seed=42)
-        out_graph = retworkx.networkx_converter(g)
-        self.assertIsInstance(out_graph, retworkx.PyGraph)
-        self.assertEqual(out_graph.nodes(), list(g.nodes))
-        self.assertEqual(
-            out_graph.weighted_edge_list(), list(g.edges(data=True))
-        )
-        self.assertEqual(out_graph.multigraph, g.is_multigraph())
+        for keep_attributes in [True, False]:
+            out_graph = retworkx.networkx_converter(
+                g, keep_attributes=keep_attributes
+            )
+            self.assertIsInstance(out_graph, retworkx.PyGraph)
+            self.assertEqual(list(out_graph.node_indexes()), list(g.nodes))
+            self.assertEqual(
+                out_graph.weighted_edge_list(), list(g.edges(data=True))
+            )
+            self.assertEqual(out_graph.multigraph, g.is_multigraph())
```

Comment:
What do you think about using a `subTest` here (and in all the other test methods) for this? Something like:

```suggestion
        for keep_attributes in [True, False]:
            with self.subTest(keep_attributes=keep_attributes):
                out_graph = retworkx.networkx_converter(
                    g, keep_attributes=keep_attributes
                )
                self.assertIsInstance(out_graph, retworkx.PyGraph)
                self.assertEqual(list(out_graph.node_indexes()), list(g.nodes))
                self.assertEqual(
                    out_graph.weighted_edge_list(), list(g.edges(data=True))
                )
                self.assertEqual(out_graph.multigraph, g.is_multigraph())
```

This would let us see which variant was running in the error output. My other thought would be to use `ddt` to parameterize the methods, but that would require adding it to the test requirements in the tox.ini, so a `subTest` is probably simpler.

Reply:
Yeah -- I was thinking of ways to make parameterize the tests more prettily, but I didn't want to add more dependencies. `subTest` seems like it'd work reasonably well for now.